src/V3LinkInc.cpp

```diff
@@ -86,6 +88,13 @@ class LinkIncVisitor final : public VNVisitor {
         // In a while condition, the statement also needs to go on the
         // back-edge to the loop header, 'incsp' is that place.
         if (m_inWhileCondp) m_inWhileCondp->addIncsp(newp->cloneTreePure(true));
+        if (m_inAssertPasssp) {
+            m_inAssertPasssp->addPasssp(newp->cloneTreePure(true));
+        } else if (m_inAssertIntrinsicFailsp) {
+            m_inAssertIntrinsicFailsp->addFailsp(newp->cloneTreePure(true));
+        } else if (m_inAssertFailsp) {
+            m_inAssertFailsp->addFailsp(newp->cloneTreePure(true));
+        }
         m_insStmtp->addHereThisAsNext(newp);
```

Comment:
AstAssert shouldn't need to be special, though I might be missing something.  Why did addHereThisAsNext not put the new statements in the right place (where the ++ is)?

Reply:
The AstAssert needs a special treatment as all statements used in pass/fail sections need to be inside the AstAssert node to be handled correctly later by V3Assert.
This error is the same as here: https://github.com/verilator/verilator/pull/3659#issuecomment-1270199816
Now, I'm extending error handling to pass/fail bodies as well.